patches/server/1052-add-number-format-api.patch

```diff
@@ -0,0 +1,162 @@
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/util/PaperScoreboardFormat.java
+@@ -0,0 +1,43 @@
++package io.papermc.paper.util;
++
++
++import io.papermc.paper.adventure.AdventureCodecs;
++import io.papermc.paper.adventure.PaperAdventure;
++import io.papermc.paper.scoreboard.numbers.NumberFormat;
++import net.minecraft.Util;
++import net.minecraft.nbt.NbtOps;
++import net.minecraft.nbt.Tag;
++
++public class PaperScoreboardFormat {
++
++    private PaperScoreboardFormat() {}
++
++
++    public static net.minecraft.network.chat.numbers.NumberFormat asVanilla(NumberFormat format) {
++        net.minecraft.network.chat.numbers.NumberFormat vanilla;
++        if (format instanceof io.papermc.paper.scoreboard.numbers.StyledFormat styled) {
++            vanilla = new net.minecraft.network.chat.numbers.StyledFormat(PaperAdventure.asVanilla(styled.getStyle()));
++        } else if (format instanceof io.papermc.paper.scoreboard.numbers.FixedFormat fixed) {
++            vanilla = new net.minecraft.network.chat.numbers.FixedFormat(io.papermc.paper.adventure.PaperAdventure
++                .asVanilla(fixed.getComponent()));
++        } else if(format.equals(NumberFormat.blank())) {
++            vanilla = net.minecraft.network.chat.numbers.BlankFormat.INSTANCE;
++        } else {
++            throw new IllegalArgumentException("Unknown format type " + format.getClass());
++        }
++
++        return vanilla;
++    }
++
++    public static NumberFormat asPaper(net.minecraft.network.chat.numbers.NumberFormat vanilla) {
++        if (vanilla instanceof net.minecraft.network.chat.numbers.StyledFormat styled) {
++            return NumberFormat.styled(PaperAdventure.asAdventure(styled.style));
++        } else if (vanilla instanceof net.minecraft.network.chat.numbers.FixedFormat fixed) {
++            return NumberFormat.fixed(io.papermc.paper.adventure.PaperAdventure.asAdventure(fixed.value));
++        } else if(vanilla instanceof net.minecraft.network.chat.numbers.BlankFormat) {
++            return NumberFormat.blank();
++        }
++
++        throw new IllegalArgumentException("Unknown format type " + vanilla.getClass());
++    }
++}
+diff --git a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
+index 3157f3d2f9ce7af4a763203672817a7f5c7bd4fb..bdd2ed0aa62eea87d356d2229bb7fe41ba293e1d 100644
+--- a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
++++ b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
+@@ -152,6 +152,30 @@ final class CraftObjective extends CraftScoreboardComponent implements Objective
+     }
+     // Paper end
+ 
++    //Paper start - add number format
++    @Override
++    public io.papermc.paper.scoreboard.numbers.NumberFormat getNumberFormat() {
++        net.minecraft.network.chat.numbers.NumberFormat vanilla = this.objective.numberFormat();
```

Comment:
This should have a `this.checkState` at the top to ensure the objective is still registered. Those exist on the other getters like getRenderType and getDisplaySlot